### ISSUE TITLE
Test suite fails for lucid 2.9.12.1

### ISSUE THREAD

**Issue by sternenseemann** [chrisdone/lucid#123]
* x86_64-linux, NixOS
* GHC 8.10.7
* Built from hackage sdist

```
Running 1 test suites...
Test suite test: RUNNING...

text
  simple
  escaping
  unicode
elements
  simple
  escaping
  unicode
  nesting
  empty
  mixed
  no closing
  no closing
attributes
  simple
  escaping
  unicode
  nesting
  empty
  mixed
  no closing
  multiple
  encoded
  nesting attributes
attributes-with
  simple
  escaping
  unicode
  nesting
  empty
  mixed FAILED [1]
  no closing
  multiple
  encoded
  nesting attributes
extension
  bootstrap
  bootstrap-attributes-extended
special-elements
  script
  style
self-closing
  br
  hr
  input
  input
commuteHtmlT
  makes using inner monads easy
monadFix
  mdo

Failures:

  test/Main.hs:43:27: 
  1) attributes-with mixed
       "<p class=\"foo\" style=\"attrib\"><style></style></p>" is not equal to any of ["<p style=\"attrib\" class=\"foo\"><style></style></p>","<pclass=\"foo\" style=\"attrib\"><style></style></p>"]

  To rerun use: --match "/attributes-with/mixed/"

Randomized with seed 1474063285

Finished in 0.0011 seconds
```

**Comment by sternenseemann** [comment]
This triggered by upgrading `unordered-containers-0.2.15.0` to `unordered-containers-0.2.16.0`, as done in Stackage LTS 18.18 -> 18.19.

**Comment by chrisdone** [comment]
Try out lucid-2.11.0

**Comment by sternenseemann** [comment]
Does the fact that this was fixed by making the order stable using `Map` (<https://github.com/chrisdone/lucid/pull/113>) indicate that the ordering is important for lucid?

In that case: Would it be possible to backport the patch or does it change the public API? 2.9.12.1 is in wide-enough use by virtue of being shipped by Stackage LTS, so a 2.9.12.2 backport release resolving this issue would be nice.
